main.py

```diff
@@ -66,6 +79,132 @@ def test_wrapper(args: Namespace) -> None:
         plot_predictions(details['Percent'], args.dataset)
 
 
+def make_predict_metadata(tile_path: str) -> List[Tuple[str, str]]:
+    predict_metadata = []
+    vh_vv_list = os.listdir(tile_path)
+    vh_vv_list = natsorted(vh_vv_list)
+    midpoint = int(len(vh_vv_list)/2)
+    vh_list = vh_vv_list[0:midpoint]
+    for i in range(midpoint):
+        vh_list[i] = os.path.join(tile_path, vh_list[i])
+    vv_list = vh_vv_list[midpoint:]
+    for i in range(midpoint):
+        vv_list[i] = os.path.join(tile_path, vv_list[i])
+    predict_metadata = tuple(zip(vh_list, vv_list))
+    return predict_metadata
+
+
+def prediction_wrapper(args: Namespace) -> None:
+    # local vars
+    mxm_tile_size = 512
+    input_sar_path = args.dataset
+    sar_name = Path(input_sar_path).stem
+    prediction_workspace_path = os.path.join(os.getcwd(),
+                                             'datasets',
+                                             'prediction_workspace')
+    sar_tiles_path = os.path.join(prediction_workspace_path, 'sar_tiles')
+    prediction_tiles_path = os.path.join(prediction_workspace_path,
+                                         'prediction_tiles')
+
+    # check there are only 2 tifs in dir
+    count = 0
+    for f in os.listdir(input_sar_path):
+        error = 'This dir should only contain the vh vv sar granules'
+        count += 1
+        if count > 2:
+            print(error)
+            exit()
+        if not f.endswith('.tif'):
+            print(error)
+            exit()
+
+    # make sar_tiles, prediction_tiles dir
+    if os.path.exists(sar_tiles_path):
+        shutil.rmtree(sar_tiles_path)
+    os.mkdir(sar_tiles_path)
+    if os.path.exists(prediction_tiles_path):
+        shutil.rmtree(prediction_tiles_path)
+    os.mkdir(prediction_tiles_path)
+
+    # tile vv vh and mv to tiles dir
+    for sar in os.listdir(input_sar_path):
+        if not os.path.isdir(os.path.join(input_sar_path, sar)):
+            tif = os.path.join(input_sar_path, sar)
+            tifData = gdal.Open(tif)
+            xStep, yStep = mxm_tile_size, mxm_tile_size
+            xSize, ySize = tifData.RasterXSize, tifData.RasterYSize
+            count = 0
+            for x in range(0, xSize, xStep):
+                for y in range(0, ySize, yStep):
+                    # fileName = '<Image|Mask>_<sar>_<0-9+>.tif'
```

Comment:
If this like isn't needed we might want to delete it instead of commenting it out.